activemodel/lib/active_model/errors.rb

```diff
@@ -487,6 +495,11 @@ def generate_message(attribute, type = :invalid, options = {})
             :"#{i18n_scope}.errors.models.#{klass.model_name.i18n_key}.#{type}" ]
         end
         defaults << :"#{i18n_scope}.errors.messages.#{type}"
+
+        catch(:exception) do
+          translation = I18n.translate(defaults.first, options.merge(default: defaults.drop(1), throw: true))
+          return translation unless translation.nil?
```

Comment:
This deep nested return seems off. Can we do a larger restructuring of this method that makes this flow clearer?

Reply:
I've tried to find a cleaner way to do this, but I can't find a clean way to keep the two following in the case were the entity responds to `:i18n_scope`:

1. Try the more precise translations first.
2. Still show `#{i18n_scope}.errors.models.#{klass.model_name.i18n_key}.attributes.#{attribute}.#{type}` as the key that failed translating when both lookup fail.

That makes it so that when `i18_scope` I need to do two things: add new defaults, and try a first lookup. That makes it hard to refactor into a private method.

I'm open to any suggestions though!